geometry-schema.json

```diff
@@ -1,91 +1,224 @@
-    "title": "geometry",
-    "description": "One geometry as defined by GeoJSON. Licensed as per original source is https://github.com/fge/sample-json-schemas/blob/master/geojson/geometry.json",
-    "type": "object",
-    "required": [ "type", "coordinates" ],
-    "oneOf": [
-        {
-            "title": "Point",
-            "properties": {
-                "type": { "enum": [ "Point" ] },
-                "coordinates": { "$ref": "#/definitions/position" }
-            }
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://geojson.org/schema/Geometry.json",
+  "title": "GeoJSON Geometry",
+  "oneOf": [
+    {
+      "$id": "#Point",
+      "title": "GeoJSON Point",
+      "type": "object",
+      "required": [
+        "type",
+        "coordinates"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "Point"
+          ]
         },
-        {
-            "title": "MultiPoint",
-            "properties": {
-                "type": { "enum": [ "MultiPoint" ] },
-                "coordinates": { "$ref": "#/definitions/positionArray" }
-            }
+        "coordinates": {
+          "type": "array",
+          "minItems": 2,
+          "items": {
+            "type": "number"
+          }
         },
-        {
-            "title": "LineString",
-            "properties": {
-                "type": { "enum": [ "LineString" ] },
-                "coordinates": { "$ref": "#/definitions/lineString" }
-            }
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#LineString",
+      "title": "GeoJSON LineString",
+      "type": "object",
+      "required": [
+        "type",
+        "coordinates"
+      ],
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "LineString"
+          ]
         },
-        {
-            "title": "MultiLineString",
-            "properties": {
-                "type": { "enum": [ "MultiLineString" ] },
-                "coordinates": {
-                    "type": "array",
-                    "items": { "$ref": "#/definitions/lineString" }
-                }
+        "coordinates": {
+          "type": "array",
+          "minItems": 2,
+          "items": {
+            "type": "array",
+            "minItems": 2,
+            "items": {
+              "type": "number"
             }
+          }
         },
-        {
-            "title": "Polygon",
-            "properties": {
-                "type": { "enum": [ "Polygon" ] },
-                "coordinates": { "$ref": "#/definitions/polygon" }
-            }
+        "bbox": {
+          "type": "array",
+          "minItems": 4,
+          "items": {
+            "type": "number"
+          }
+        }
+      }
+    },
+    {
+      "$id": "#Polygon",
```

Comment:
internal $id is not used in the official schema, the rather provide a specific document per type to point directly to it.